code/__HELPERS/colors.dm

```diff
@@ -148,3 +148,69 @@
 			return "#[num2hex(c, 2)][num2hex(m, 2)][num2hex(x, 2)]"
 
 #define RANDOM_COLOUR (rgb(rand(0,255),rand(0,255),rand(0,255)))
+
+/* Generates an HSL color transition matrix filter which nicely paints an object
+ * without making it a deep fried blob of color
+ * saturation_behavior determines how we handle color saturation:
+ * SATURATION_MULTIPLY - Multiply pixel's saturation by color's saturation. Paints accents while keeping dim areas dim.
+ * SATURATION_OVERRIDE- Affects original lightness/saturation to ensure that pale objects still get doused in color
+ */
+/proc/color_transition_filter(new_color, saturation_behavior = SATURATION_MULTIPLY)
+	if (islist(new_color))
+		new_color = rgb(new_color[1], new_color[2], new_color[3])
+	new_color = rgb2num(new_color, COLORSPACE_HSL)
+	var/hue = new_color[1] / 360
+	var/saturation = new_color[2] / 100
+	var/added_saturation = 0
+	var/deducted_light = 0
+	if (saturation_behavior == SATURATION_OVERRIDE)
+		added_saturation = saturation * 0.75
+		deducted_light = saturation * 0.5
+		saturation = min(saturation, 1 - added_saturation)
+
+	var/list/new_matrix = list(
+		0, 0, 0, 0, // Ignore original hue
+		0, saturation, 0, 0, // Multiply the saturation by ours
+		0, 0, 1 - deducted_light, 0, // If we're highly saturated then remove a bit of lightness to keep some color in
+		0, 0, 0, 1, // Preserve alpha
+		hue, added_saturation, 0, 0, // And apply our preferred hue and some saturation if we're oversaturated
+	)
+	return color_matrix_filter(new_matrix, FILTER_COLOR_HSL)
+
+/// Applies a color filter to a hex/RGB list color
+/proc/apply_color_filter(color, list/matrix, colorspace = COLORSPACE_HSL)
+	if (istext(color))
+		color = rgb2num(color)
+	// Pad alpha if we're lacking it
+	if (length(color) < 4)
+		color += 255
+	// Ugly but this is the easiest way
+	color = rgb(color[1], color[2], color[3], color[4], space = colorspace)
+	color = rgb2num(color, colorspace)
+
+	var/row_length = 3
+	if (length(matrix) >= 16)
+		row_length = 4
+
+	var/list/new_color = list(0, 0, 0, 0)
+
+	for (var/row_index in 1 to length(matrix) / row_length)
+		for (var/row_elem in 1 to row_length)
+			var/elem = matrix[(row_index - 1) * row_length + row_elem]
+			if (row_index != (length(matrix) / row_length) || (length(matrix) != 12 && length(matrix) != 20))
+				new_color[row_index] += color[row_elem] * elem
+				continue
+
+			// Constant values at the end of the list (if we have such)
+			var/multiplier = 255
+			if (colorspace != COLORSPACE_RGB)
+				// Hue goes from 0 to 360
+				if (row_elem == 1)
+					multiplier = 360
+				// Value, luminance, chroma, etc go from 0 to 100
+				else
+					multiplier = 100
+			new_color[row_elem] += elem * multiplier
```

Comment:
would prefer you to be explicit in case of fun new colorspaces in future (idk man just in case)